recipes/luajit/all/conanfile.py

```diff
@@ -83,18 +88,26 @@ def _patch_sources(self):
                 replace_in_file(self, makefile,
                                       'TARGET_O= $(LUAJIT_A)',
                                       'TARGET_O= $(LUAJIT_SO)')
-            if "clang" in str(self.settings.compiler):
-                replace_in_file(self, makefile, 'CC= $(DEFAULT_CC)', 'CC= clang')
 
-    @property
-    def _macosx_deployment_target(self):
-        return self.settings.get_safe("os.version")
+    def _apple_deployment_target(self, default=None):
+        return self.settings.get_safe("os.version", default=default)
 
     @property
     def _make_arguments(self):
         args = [f"PREFIX={unix_path(self, self.package_folder)}"]
-        if is_apple_os(self) and self._macosx_deployment_target:
-            args.append(f"MACOSX_DEPLOYMENT_TARGET={self._macosx_deployment_target}")
+        if "clang" in str(self.settings.compiler):
+            args.append("DEFAULT_CC=clang")
```

Comment:
This is incorrect and basically reproduces the existing sloppy handling of CC in the recipe. It hackily replaces the default `CC=gcc` with `clang`, but it ignores the exact compiler executable set in the Conan profile and won't work for any other compilers in general.

It should really be something like
```python
cc = AutotoolsToolchain(self).vars()["CC"]
args.append(f"CC={cc}")
```
Similarly, `HOST_CC` needs to be set to the `CC_FOR_BUILD` value from the toolchain vars. Something like https://github.com/conan-io/conan/issues/17602 would be very welcome here.

Reply:
that's what iOS build docs suggest to do. So should I add this variable only for iOS then and revert the general compiler adjustment?